node/src/components/storage.rs

```diff
@@ -307,6 +308,22 @@ impl Storage {
         drop(cursor);
         drop(block_txn);
 
+        // Check the integrity of the block body database.
+        check_block_body_db(&env, &block_body_db).unwrap_or_else(|error| {
+            panic!(
+                "Error in the integrity check of the block_body database: {:?}",
+                error
+            )
+        });
+
+        // Check the integrity of the block metadata database.
+        check_block_metadata_db(&env, &block_metadata_db).unwrap_or_else(|error| {
+            panic!(
+                "Error in the integrity check of the block_metadata database: {:?}",
+                error
+            )
```

Comment:
I am not sure we should be the ones deciding to panic. Can't you simply call `?` on the method and pass on the error to the caller of the constructor?

Reply:
Aaah yes we could

Reply:
Changed to use `?`